Solaris shadow.h definition is:
extern struct spwd *fgetspent_r(FILE *, struct spwd *, char *, int);

This is due to a difference between the Solaris & Linux definitions of
fgetspent_r - Solaris returns a pointer on success & NULL on failure,
Linux returns 0 on success and -1 for failure.

Proper fix should be sent to upstream.

--- accountsservice-0.6.55/src/daemon.c
+++ accountsservice-0.6.55/src/daemon.c
@@ -197,12 +197,12 @@ }
                 shadow_users = g_hash_table_new_full (g_str_hash, g_str_equal, g_free, g_free);
 
                 do {
-                        int ret = 0;
+                        struct spwd *ret = NULL;
 
                         shadow_entry_buffers = g_malloc0 (sizeof (*shadow_entry_buffers));
 
-                        ret = fgetspent_r (fp, &shadow_entry_buffers->spbuf, shadow_entry_buffers->buf, sizeof (shadow_entry_buffers->buf), &shadow_entry);
-                        if (ret == 0) {
+                        ret = fgetspent_r (fp, &shadow_entry_buffers->spbuf, shadow_entry_buffers->buf, sizeof (shadow_entry_buffers->buf));
+                        if (ret != NULL) {
                                 g_hash_table_insert (shadow_users, g_strdup (shadow_entry->sp_namp), shadow_entry_buffers);
                         } else {
                                 g_free (shadow_entry_buffers);
